flytekit/lazy_import/lazy_module.py

```diff
@@ -24,6 +34,10 @@ def lazy_module(fullname):
         return sys.modules[fullname]
     # https://docs.python.org/3/library/importlib.html#implementing-lazy-imports
     spec = importlib.util.find_spec(fullname)
+    if spec is None:
+        # Return a dummy module if the module is not found in the python environment,
+        # so that we can raise a proper error when the user tries to access an attribute in the module.
+        return DummyModule(fullname)
```

Comment:
As a suggestion, a test like this could cover all the new lines in this file:

```python
    def test_lazy_module_not_found(self) -> None:
        """Test that accessing an attribute on a nonexistent, lazily loaded module raises an ImportError."""
        nonexistent_lazy_module = lazy_module("nonexistent_module")
        with pytest.raises(ImportError):
            nonexistent_lazy_module.attr
```